### PR TITLE
feature: Add dark mode support and Refactor UI

### LINKED ISSUE #100: UI: Add dark mode support and Refactor UI
Implement support for dark mode in the app and configure color schemes in the theme for both light and dark modes. Refactor UI components to replace the hardcoded colors, ensuring they utilize theme-defined colors instead.

### PR DESCRIPTION
**Pull request by aaisel**
# Description

This PR adds support for dark mode in the app and configures the color schemes found in the `Theme.kt`  file for both light and dark modes. It also refactors UI components to replace any hardcoded colors, ensuring they utilize theme-defined colors instead. Additionally, text fields in the app have been changed to non-capital letters for a more cohesive and modern design.

Color scheme used:

#00A5DF

Fixes #100 

# Preview

`SignInScreen` Light Mode | `OverviewScreen` Light Mode | `SignInScreen` Dark Mode | `OverviewScreen` Dark Mode 
:-------------------------:|:-------------------------:|:-------------------------:|:-------------------------:
Screenshot_20241104-144252_SampleApp | Screenshot_20241108-115945_SampleApp | Screenshot_20241104-144218_SampleApp | Screenshot_20241108-115929_SampleApp

# How Has This Been Tested?

Tests are already implemented. `EndToEndTest` and `AddNoteTest` have been updated to match with the changes.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged in downstream modules, no conflicts with main

### CODE REVIEW

File: app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt [hunk 1]
```diff
@@ -72,7 +73,7 @@ fun AddNoteScreen(
   var className by remember { mutableStateOf("") }
   var classCode by remember { mutableStateOf("") }
   var classYear by remember { mutableIntStateOf(currentYear) }
-  var template by remember { mutableStateOf("Choose An Option") }
+  var template by remember { mutableStateOf("Choose an option") }
```
Comment (AvillegasH) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt:76]: Here I am not sure if we should stick to lower or capital letters at the start of each word. Maybe we should discuss this in a meeting with all team members

**Review by AvillegasH — APPROVED** [review]
Not much to say, very nice addition to the app. The code was well written. Maybe we could add even more colors for future sprints.

File: app/src/main/java/com/github/onlynotesswent/ui/overview/Overview.kt [hunk 2]
```diff
@@ -236,7 +236,7 @@ fun NoteItem(note: Note, onClick: () -> Unit) {
               fontWeight = FontWeight.Bold,
               color = MaterialTheme.colorScheme.onPrimaryContainer)
           Text(
-              text = note.noteClass.classCode + " " + note.noteClass.className,
+              text = note.noteClass.classCode,
```
Comment (AvillegasH) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/Overview.kt:239]: Maybe here having both `classCode` and `className`  was better, but I dont know if they both fit on the `noteCard`

File: app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt [hunk 3]
```diff
@@ -116,21 +116,17 @@ fun AddNoteScreen(
             verticalArrangement = Arrangement.spacedBy(8.dp)) {
               Spacer(modifier = Modifier.height(30.dp))
 
-              OutlinedTextField(
+              NoteDataTextField(
```
Comment (AvillegasH) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt:119]: Good addition, we will have to then put this common composable in a separate class

File: app/src/main/java/com/github/onlynotesswent/ui/theme/Theme.kt [hunk 4]
```diff
@@ -10,34 +10,48 @@ import androidx.compose.material3.dynamicLightColorScheme
...
+        onTertiary = LightShade,
+        onBackground = DarkShade,
+        onSurface = DarkShade)
 
 @Composable
 fun AppTheme(
     darkTheme: Boolean = isSystemInDarkTheme(),
-    // Dynamic color is available on Android 12+
-    dynamicColor: Boolean = true,
+    dynamicColor: Boolean = false, // disabled to ensure consistent colors across devices
```
Comment (AvillegasH) [inline app/src/main/java/com/github/onlynotesswent/ui/theme/Theme.kt:54]: Do we have to put it to false  as of right now?

Comment (AvillegasH) [reply app/src/main/java/com/github/onlynotesswent/ui/theme/Theme.kt:54]: Never mind, I got it now

**Review by AvillegasH — APPROVED** [review]
New additions were good, specially the `NoteDataTextField`. Good job

File: app/src/main/java/com/github/onlynotesswent/ui/overview/Overview.kt (hunk 2 above)
Comment (aaisel) [reply app/src/main/java/com/github/onlynotesswent/ui/overview/Overview.kt:239]: With the name it's too long, so for now let's just save the class code and see later if we decide to modify it

File: app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt [hunk 5]
```diff
@@ -303,3 +281,37 @@ fun NoteDataTextField(
...
+ * A composable function that displays the top app bar for the screen. It is composed of a back
+ * button and a title.
+ *
+ * @param title The title to be displayed in the top app bar.
+ * @param titleTestTag The test tag for the title.
+ * @param onBackClick The callback to be invoked when the back button is clicked.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ScreenTopBar(title: String, titleTestTag: String, onBackClick: () -> Unit) {
```
Comment (AvillegasH) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt:295]: Nice addition, we will also move this to a separate common composable class

File: app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt [hunk 6]
```diff
@@ -303,3 +281,37 @@ fun NoteDataTextField(
...
+          Text(
+              title,
+              color = MaterialTheme.colorScheme.onSurface,
+              modifier = Modifier.testTag(titleTestTag))
+          Spacer(modifier = Modifier.weight(2f))
+        }
+      },
+      navigationIcon = {
+        IconButton(onClick = onBackClick, Modifier.testTag("goBackButton")) {
+          Icon(
```
Comment (AvillegasH) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt:311]: Also for more modularity, we should have the option to define the Icon and not have its value hard coded

**Review by AvillegasH — CHANGES_REQUESTED** [review]
Good additions. You should try to modularize more the top bar by passing the Icon as an argument instead of having the hard coded value. Other than that, the additions were well documented.

File: app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt (hunk 6 above)
Comment (aaisel) [reply app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt:311]: Yeah, you're right, thanks

**Review by AvillegasH — APPROVED** [review]

**Pull request merged**